modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/infrastructure/snapshot/CurrencySnapshotCreator.scala

```diff
@@ -154,6 +157,16 @@ object CurrencySnapshotCreator {
                     .getOrElse(SortedSet.empty[RewardTransaction].pure[F])
               )
 
+          rejectedBlockEvents = NonEmptyList.fromListUnsafe(acceptanceResult.notAccepted.collect {
+            case (b, r: BlockRejectionReason) => (BlockEvent(b), Some(r))
+          })
+          _ <- currencySnapshotValidationErrorStorage.addValidationErrors(rejectedBlockEvents)
+
+          rejectedMessageEvents = NonEmptyList.fromListUnsafe(
+            messagesAcceptanceResult.notAccepted.map(m => (CurrencyMessageEvent(m), None))
+          )
+          _ <- currencySnapshotValidationErrorStorage.addValidationErrors(rejectedMessageEvents)
```

Comment:
you ignore messages in the encoder, so why you add them here? is the Encoder invalid then?